jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/ProcessServiceImpl.java

```diff
@@ -295,6 +295,43 @@ public void signalProcessInstances(String deploymentId, List<Long> processInstan
 
     }
 
+    @Override
+    public void signalProcessInstanceByCorrelationKey(CorrelationKey correlationKey, String signalName, Object event) {
+        ProcessInstanceDesc pi = dataService.getProcessInstanceByCorrelationKey(correlationKey);
+        if(pi == null) {
+            throw new ProcessInstanceNotFoundException("Process with correlation key " + correlationKey + " not found");
+        }
+        signalProcessInstance(pi.getId(), signalName, event);
+        
+    }
+
+    @Override
+    public void signalProcessInstanceByCorrelationKey(String deploymentId,
+                                                      CorrelationKey correlationKey,
+                                                      String signalName,
+                                                      Object event) {
+        ProcessInstanceDesc pi = dataService.getProcessInstanceByCorrelationKey(correlationKey);
+        if(pi == null) {
+            throw new ProcessInstanceNotFoundException("Process with correlation key " + correlationKey + " not found");
+        }
+        signalProcessInstance(deploymentId, pi.getId(), signalName, event);
+    }
+
+    @Override
+    public void signalProcessInstancesByCorrelationKeys(List<CorrelationKey> correlationKeys, String signalName, Object event) {
+        correlationKeys.forEach(key -> signalProcessInstanceByCorrelationKey(key, signalName, event));
```

Comment:
`correlationKeys` might be `null` - maybe it's worth testing it before calling `forEach`

Reply:
ok